### PR TITLE
Remove validation headers file

### LINKED ISSUE #201: Validation of animal/sample files throws error regarding missing headers definition file
### Problem

Attempting to validate an animal/sample file results in the following error:
```
The following issue(s) with the file(s) were found:
Please address the issues below in your file and try again. If it is not clear what the issue is, you may proceed to the submission form and a curator will attempt to resolve the issue.
FAILED: animals_samples_treatments.xlsx
animals_samples_treatments.xlsx: [Errno 2] No such file or directory: 'DataRepo/example_data/sample_and_animal_tables_headers.yaml'
```
image

### Steps to reproduce

1. Upload an animal/sample file (*e.g.* [`data_submission_animal_sample_table.xlsx`](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/DataRepo/example_data/data_submission_animal_sample_table.xlsx)
2. Click Validate

### Current behavior

```
[Errno 2] No such file or directory: 'DataRepo/example_data/sample_and_animal_tables_headers.yaml'
```

### Expected behavior

File should validate with no errors.

### Suggested Change

The validation should use the default headers and not specify a custom headers file.

Perhaps removing this line:
https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/f3589e84c3205a6e7b833d4c1754ddf0549d0b48/DataRepo/views.py#L1087

### Comment

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will not fix
- E.g. Getting edge case X to work requires too much effort and will not be
  fixed in this effort.

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the fix.

#### Tests

A test should be planned for each requirement (above), where possible.

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Eliminates the header yaml file from the validation process and fixes a couple other issues that came up:

- The missing headers logic in utils.py had a problem.  I'd encountered an [error](https://github.com/Princeton-LSI-ResearchComputing/tracebase/issues/201#issuecomment-954223284) about missing optional headers (headers that weren't in the file being validated).
- I added a catch of an exception generated by an assert that's used to abort database transactions in debug mode.

## Affected Issue Numbers

- Resolves #201

## Code Review Notes

I'd been unable to intentionally reproduce the missing headers error.  It may have had to do with my database and ignored exceptions, but I didn't think it was worth it to figure it out.

The "Debugging" exception catch wasn't encountered before because I was only checking the researcher exception and marking it as a warning.

## Checklist

- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [ ] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

**Review by lparsons — APPROVED** [review]
This looks good to me, though I did have one minor suggestion and a question about the Exception handling (mostly for my own edification and possible clarification in the comments).

**Review by jcmatese — APPROVED** [review]

**Review by fkang-pu — APPROVED** [review]

File: DataRepo/utils.py [hunk 1]
```diff
@@ -120,10 +120,25 @@ def validate_researcher(self, data):
 
     def load_sample_table(self, data, skip_researcher_check=False, debug=False):
         self.debug = debug
-        self.validate_sample_table(data, skip_researcher_check=skip_researcher_check)
+
+        # Create an array to hold the csv reader data so that iterations from validating doesn't leave the csv reader
+        # empty/at-the-end upon the import loop
+        array_data = []
         for row in data:
+            array_data.append(row)
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/utils.py:128]: I think using `list` is definitely better.  And I'll update the variable name.  Original was `data`.
- [ ] Change from loop to `sample_table_data = list(data)`

File: DataRepo/views.py [hunk 2]
```diff
@@ -989,9 +987,17 @@ def validate_load_files(self, animal_sample_dict, accucor_dict, ash_yaml):
             )
             results[animal_sample_name] = "WARNING"
         except Exception as e:
-            valid = False
-            errors[animal_sample_name].append(f"{animal_sample_name}: {str(e)}")
-            results[animal_sample_name] = "FAILED"
+            estr = str(e)
+            # We are using the presence of the string "Debugging..." to infer that it got to the end of the load
+            # without an exception.  If there is no "Debugging" message, then an exception did not occur anyway
+            if "Debugging" not in estr:
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/views.py:993]: I had the same thought about fragility.  That wasn't new in this PR, though I did touch the line.  I couldn't remember what was going on, so when I re-figured it out, I added some comments to the code in `utils.py` (and here? - not enough lines of code in this snippet for context.  The exception is an `AssertionError` (sp?), but `AssertionError` was also used elsewhere in the code, hence matching the string was the only option aside from changing John's code.

As I understand John's loading code, the exception is intentional to abort the database transaction in debug mode by causing an exception when an assert on debug is true.  I'm not sure what other ways there would be to abort database transactions, but if we stay with throwing an exception, we should probably throw a custom exception that can be caught stably in my code.  What do you think?

[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/views.py:993]: Now looking at the full diff, it is indeed new in this location, but it exists elsewhere as well.  I just wasn't encountering the need for it here due to the CSV reader bug I fixed.

[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/views.py:993]: Just for reference... 1016 was added as well.  But lines 1063 & 1081 were already there.

[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/views.py:993]: Created issue #236 to address this.

**Pull request merged**